search/includes/classes/class-queue.php

```diff
@@ -313,25 +316,22 @@ public function queue_object( $object_id, $object_type = 'post', $options = arra
 
 		$index_version = $this->get_index_version_number_from_options( $object_type, $options );
 
-		$table_name = $this->schema->get_table_name();
+		$priority = (int) ( $options['priority'] ?? self::INDEX_DEFAULT_PRIORITY );
 
-		// Have to escape this separately so we can insert NULL if no start time specified
-		$start_time_escaped = $next_index_time ? $wpdb->prepare( '%s', array( $next_index_time ) ) : 'NULL';
+		$table_name = $this->schema->get_table_name();
 
 		$original_suppress = $wpdb->suppress_errors;
 
 		// Suppress errors because duplicate key errors are expected, as we use the UNIQUE index
 		// to de-duplicate queued jobs without first querying to see if the object is queued
 		$wpdb->suppress_errors( true );
 
-		// phpcs:ignore WordPress.DB.DirectDatabaseQuery
+		// phpcs:ignore WordPress.DB
 		$wpdb->query(
-			$wpdb->prepare(
-				"INSERT INTO $table_name ( `object_id`, `object_type`, `start_time`, `status`, `index_version` ) VALUES ( %d, %s, {$start_time_escaped}, %s, %d )", // Cannot prepare table name. @codingStandardsIgnoreLine
-				$object_id,
-				$object_type,
-				'queued',
-				$index_version
+			// phpcs:ignore WordPress.DB
+			$wpdb->prepare( "INSERT INTO {$table_name} (object_id, object_type, status, index_version, start_time, priority)
+				VALUES (%d, %s, 'queued', %d, %s, %d) ON DUPLICATE KEY UPDATE priority = LEAST(priority, %d)",
+				[ $object_id, $object_type, $index_version, $next_index_time, $priority, $priority ]
```

Comment:
Using the placeholder in `LEAST` because otherwise, the monkeys will finally take over the zoo:
* In MariaDB until 10.3.2, you have to use `LEAST(priority, VALUES(priority))`;
* In MariaDB since 10.3.3, it will be `LEAST(priority, VALUE(priority))`;
* In MySQL since 8.0.20, it should be `LEAST(priority, new.priority)`

As always, this will break SUDDENLY.